datafusion/core/tests/sqllogictests/test_files/join_disable_repartition_joins.slt

```diff
@@ -0,0 +1,30 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+##########
+## Join Tests
+##########
+
+# turn off repartition_joins
+statement ok
+set datafusion.optimizer.repartition_joins = false;
+
+include ./join.slt
+
+# turn on repartition_joins
+statement ok
+set datafusion.optimizer.repartition_joins = true;
```

Comment:
> sqllogictest we output the sql, which makes the output noisy and probably not very useful, I was thinking to create a ticket to ouputit file name and indented test name. the similar way as cargo test


I think this is a good idea

> do you guys think we should create a .slt file per use case like disabled repartition join? or reuse join.slt?

I like the idea of reusing join.slt but it will likely make some tests in the future harder if they vary based on the setting (e.g. plan tests)

Reply:
> do you guys think we should create a .slt file per use case like disabled repartition join? or reuse join.slt?

I think it's worth creating this file now. I first tried adding everything to `join.slt` and found a lot of duplicated code, since most tests need to be run twice.
```sql
# query x
statement ok
set datafusion.optimizer.repartition_joins = false;

query ITI rowsort
SELECT xxx
----
a a a

# switch to another config
statement ok
set datafusion.optimizer.repartition_joins = true;

query ITI rowsort
SELECT xxx
----
a a a
```

Using include reduces a lot of codes, originally the addition was `1100+`, now it is `655`.
For the tests whose setting is complex, we can also add them to `join.slt`.